src/editor/contentState/tabCtrl.js

```diff
@@ -118,25 +118,27 @@ const tabCtrl = ContentState => {
     }
 
     this.appendChild(newList, listItem)
-    this.partialRender()
+    return this.partialRender()
   }
 
-  // ContentState.prototype.insertTab = function () {
-  //   // TODO(fxha): Tabs and all spaces with length > 1 are converted to a single space
-  //                  in editor mode. Maybe write a HTML "tab" element
-  //   // TODO(fxha): create setting entry "tabSize: <int>"
-  //   const tabCharacter = new Array(4).join(' ')
-  //   const { start, end } = this.cursor
-  //   const startBlock = this.getBlock(start.key)
-  //   const endBlock = this.getBlock(end.key)
-  //   if (this.cursor.start.key === this.cursor.end.key) {
-  //     startBlock.text = startBlock.text.substring(0, start.offset) + tabCharacter + endBlock.text.substring(end.offset)
-  //     // workaround: see todo 1
-  //     this.cursor.start.offset += 1 // tabCharacter.length
-  //     this.cursor.end.offset += 1 // tabCharacter.length
-  //     this.render()
-  //   }
-  // }
+  ContentState.prototype.insertTab = function () {
+    let tabSize = this.tabSize
+    if (!tabSize || typeof tabSize !== 'number') {
+      tabSize = 4
+    } else if (tabSize < 1) {
+      tabSize = 1
+    }
+    const tabCharacter = new Array(tabSize + 1).join(' ')
+    const { start, end } = this.cursor
+    const startBlock = this.getBlock(start.key)
+    const endBlock = this.getBlock(end.key)
+    if (start.key === end.key) {
+      startBlock.text = startBlock.text.substring(0, start.offset) + tabCharacter + endBlock.text.substring(end.offset)
+      start.offset += tabCharacter.length
+      end.offset += tabCharacter.length
```

Comment:
Please write this bellow:

```javascript
this.cursor = {
    start: {},
    end: {}
}
```

We can record this action/operation in our redo/undo stack only in this way(use `this.cursor = {}`).